From 1ed87ecb7f0b8f2be4c04aed242f2114975ab8e1 Mon Sep 17 00:00:00 2001
From: Mike Bland <mbland@acm.org>
Date: Mon, 2 Oct 2017 22:02:00 -0400
Subject: [PATCH] test/bats: Add fake tput to fix Alpine tests

The test cases that use the `-p` or `--pretty` flag would fail if `tput`
wasn't available (installed as part of the `ncurses` package). On
Alpine, `ncurses` isn't installed by default. This change ensures the
tests validate the expected behavior regardless of whether `tput` is
available.
---
 test/test_helper.bash | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/test/test_helper.bash b/test/test_helper.bash
index 302f743..af89003 100644
--- a/test/test_helper.bash
+++ b/test/test_helper.bash
@@ -11,6 +11,13 @@ filter_control_sequences() {
   "$@" | sed $'s,\x1b\\[[0-9;]*[a-zA-Z],,g'
 }
 
+if ! command -v tput >/dev/null; then
+  tput() {
+    printf '1000\n'
+  }
+  export -f tput
+fi
+
 teardown() {
   [ -d "$TMP" ] && rm -f "$TMP"/*
 }
-- 
2.15.0

